api/machine/machine_driver.go

```diff
@@ -44,22 +48,78 @@ func (h *Handlers) ActionHandler(actionName string, action *types.Action, apiCon
 	return nil
 }
 
-// Formatter for MachineDriver
-func (h *Handlers) Formatter(apiContext *types.APIContext, resource *types.RawResource) {
+// FormatterDriver for MachineDriver
+func (h *Handlers) FormatterDriver(apiContext *types.APIContext, resource *types.RawResource) {
 	resource.AddAction(apiContext, "activate")
 	resource.AddAction(apiContext, "deactivate")
 }
 
 // Formatter for Machine
-func Formatter(apiContext *types.APIContext, resource *types.RawResource) {
+func (h *Handlers) Formatter(apiContext *types.APIContext, resource *types.RawResource) {
 	roles := convert.ToStringSlice(resource.Values[client.MachineFieldRole])
 	if len(roles) == 0 {
 		resource.Values[client.MachineFieldRole] = []string{"worker"}
 	}
 
-	// remove link
 	machineTemplateID, ok := resource.Values["machineTemplateId"]
-	if !ok || machineTemplateID == nil {
+
+	// check single etcd/control node
+	checkSingleNode, _ := checkSingleNode(resource, h.Clusters, h.MachineClient)
+
+	// remove link
+	if !ok || machineTemplateID == nil || checkSingleNode {
 		delete(resource.Links, "remove")
 	}
 }
+
+func checkSingleNode(resource *types.RawResource, clusters v3.ClusterInterface, machineInterface v3.MachineInterface) (bool, error) {
```

Comment:
this logic repeats the one in machine-controller. Think if we can move it out to some centralized place